### ISSUE TITLE
azurerm_virtual_machine_extension does not depend properly on azurerm_virtual_machine

### ISSUE THREAD

**Issue by bentterp** [terraform-providers/terraform-provider-azurerm#3769]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_virtual_machine_extension`
* `azurerm_virtual_machine`

### Terraform Configuration Files

```hcl
resource "azurerm_virtual_machine_extension" "monitoringagent" {
  name                 = "OmsAgentForLinux"
  location             = azurerm_resource_group.vm.location
  resource_group_name  = azurerm_resource_group.vm.name
  virtual_machine_name = azurerm_virtual_machine.vm.name
  publisher            = "Microsoft.EnterpriseCloud.Monitoring"
  type                 = "OmsAgentForLinux"
  type_handler_version = "1.11"
  auto_upgrade_minor_version = true 
  settings = jsonencode({workspaceId : var.oms_workspaceId})
  protected_settings = jsonencode({workspaceKey : var.oms_workspaceKey})
}
```

### Expected Behavior

When I taint the VM, I expect the extension to be removed and reinstalled.

### Actual Behavior

Terraform just wants to destroy/create the VM, which gives me a shiny new VM without the extension.

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by jwueste** [comment]
As a workaround, we add a tag to the VM like 
```terraform
resource "azurerm_virtual_machine" "vm" {
 lifecycle {
    ignore_changes = ["tags"]
  }
tags {
    created = "${timestamp()}"
  }
```

and use the value of this tag in the extensionname
```terraform
resource "azurerm_virtual_machine_extension" "vmExtOms" {
  depends_on = ["azurerm_virtual_machine.vm"]
  name                 = "OMS_${replace(azurerm_virtual_machine.vm.tags.created, ":", ".")}"
```

**Comment by tombuildsstuff** [comment]
👋 

Taking a look through here it appears this'll be fixed via #2456 - rather than having multiple issues open tracking the same thing I'm going to close this issue in favour of that one; would you mind subscribing to #2456 for updates?

Thanks!
